src/Symfony/Component/Lock/Store/RedisStore.php

```diff
@@ -55,13 +58,70 @@ public function __construct($redisClient, float $initialTtl = 300.0)
     public function save(Key $key)
     {
         $script = '
-            if redis.call("GET", KEYS[1]) == ARGV[1] then
-                return redis.call("PEXPIRE", KEYS[1], ARGV[2])
-            elseif redis.call("SET", KEYS[1], ARGV[1], "NX", "PX", ARGV[2]) then
-                return 1
-            else
-                return 0
+            local key = KEYS[1]
+            local uniqueToken = ARGV[1]
+            local ttl = tonumber(ARGV[2])
+
+            local now = redis.call("TIME")
```

Comment:
`TIME` is not deterministic before Redis 5.0 (see #35780 (comment))

But using php's time (multiple source/server) is not reliable.

What's the best solution @lyrixx ?

Reply:
In semaphore I used the server's time. I think it's more reliable than hopping user are running redis > 5.X

Reply:
Oh, I did not realized that the script of REDIS < 5 was broken because of that.
Thanks for the confirmation, I've refactored this part.